src/coreclr/jit/emitxarch.cpp

```diff
@@ -1280,501 +1280,354 @@ bool isPrefix(BYTE b)
 }
 
 //------------------------------------------------------------------------
-// emitOutputSimdPrefixIfNeeded: Outputs EVEX prefix (in case of AVX512 instructions),
-// VEX prefix (in case of AVX instructions) and REX.R/X/W/B otherwise.
+// emitExtractEvexPrefix: Extract the EVEX prefix
 //
 // Arguments:
-//    ins -- processor instruction to check.
-//    dst -- buffer to write prefix to.
-//    code -- opcode bits.
-//    attr -- operand size
+//    [In]      ins  -- processor instruction to check.
+//    [In, Out] code -- opcode bits which will no longer contain the evex prefix on return
 //
 // Return Value:
-//    Size of prefix.
+//    The extracted EVEX prefix.
 //
-unsigned emitter::emitOutputSimdPrefixIfNeeded(instruction ins, BYTE* dst, code_t& code)
+emitter::code_t emitter::emitExtractEvexPrefix(instruction ins, code_t& code)
```

Comment:
I initially extracted these methods as I thought the easiest thing was going to be to just pass the `code_t` through to `emitGetVexPrefixSize` and then do something like:
```cpp
code_t vexPrefix = emitExtractVexPrefix(ins, code);
assert(vexPrefix != 0);

if ((vexPrefix & 0xFFFF7F80) == 0x00C46100)
{
    return 2;
}

return 3;
```

However, there is quite a bit more logic that goes into correctly building up `code` and since we don't cache it anywhere, it was going to negatively impact throughput.

I ended up leaving the helper method here as it may still be useful in the future and it isolates a large chunk of complex logic that was just splatted inline before.